lib/src/scheduler.rs

```diff
@@ -11,3 +11,51 @@ where
     fn set_processor(&mut self, processor: &T);
     fn schedule(&mut self) -> (i32, VecDeque<NodeIndex>);
 }
+
+#[derive(Clone, Default)]
+pub struct NodeLog {
+    pub core_id: usize,
+    pub node_id: i32,
+    pub start_time: i32,
+    pub end_time: i32,
+}
+
+#[derive(Clone, Default)]
+pub struct ProcessorLog {
+    pub core_log: Vec<CoreLog>,
+    pub average_utilization_rate: f32,
+    pub variance_utilization_rate: f32,
+}
+
+impl ProcessorLog {
+    pub fn set_average_utilization_rate(&mut self) {
+        self.average_utilization_rate = self
+            .core_log
+            .iter()
+            .map(|core_data| core_data.utilization_rate)
+            .sum::<f32>()
+            / self.core_log.len() as f32;
+    }
+
+    pub fn set_variance_utilization_rate(&mut self) {
+        self.variance_utilization_rate = self
+            .core_log
+            .iter()
+            .map(|core_data| (core_data.utilization_rate - self.average_utilization_rate).powi(2))
+            .sum::<f32>()
+            / self.core_log.len() as f32;
+    }
+}
+
+#[derive(Clone, Default)]
+pub struct CoreLog {
+    pub core_id: usize,
+    pub total_proc_time: i32,
+    pub utilization_rate: f32,
+}
+
+impl CoreLog {
+    pub fn set_utilization_rate(&mut self, schedule_length: i32) {
```

Comment:
A setter is simply a method that assigns a value to a field.
https://qiita.com/takahirocook/items/27828bc8477735612021

This method is not a setter, so a name like `calculate_utilization` is preferred.

Reply:
Modified.